### ISSUE TITLE
Add support for string URL validation in models

### ISSUE THREAD

**Issue by l3ku** [lukeautry/tsoa#567]
It would be nice to have validation for URLs as part of tsoa so that for a model we could use the `@isURL()` decorator for a string attribute of a model to add extra validation (like e.g. `@isInt()`/`@maximum`/`@minimum` for numbers right now). 

## Sorting

* **I'm submitting a ...**
  - [ ] bug report
  - [x] feature request
  - [ ] support request

* I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

```ts
export interface MyModel {
  /** 
   * Example URL.
   * @isURL()
   */
  exampleURL: string;
  ...
```
Sending data that is not an URL should cause a validation error.

## Current Behavior

```ts
export interface MyModel {
  /** 
   * Example URL.
   */
  exampleURL: string;
  ...

```
Validation for the URL has to be done manually in the control or using a URL regexp in `@pattern()`.

## Pontential solution
Use the NodeJS [built-in URL module](https://nodejs.org/api/url.html) to validate the model attribute that was decorated with `isURL()`.

**Comment by dgreene1** [comment]
We already provide the `@pattern` And `@example` decorators. We won’t add this since it’s a duplicate (with minor syntactic sugar) which would open the flood gates for a new PR for every common pattern.

**Comment by WoH** [comment]
I'd add that anything beyond should only be solved using a Custom formatter proposal, see #496
